File: haohaninfo/交易回測案例/45-1.R
# 讀取檔案
I020<- read.csv('Futures_20170815_I020.csv')
# 設置資料欄位名稱
colnames(I020)<-c("INFO_TIME","MATCH_TIME","PROD","ITEM","PRICE","QTY","AMOUNT","MATCH_BUY_CNT","MATCH_SELL_CNT")
# 設定日期
Date <- "20170508"
# 取相關欄位
A01 <- I020[c(1,4,5)]
# 選取開倉及平倉時間
A02 <- subset(A01, INFO_TIME>=as.numeric(09000000) & INFO_TIME<=as.numeric(11000000))
# 設定進場時間及價格
OrderTime <- A02[1,]$INFO_TIME
OrderPrice <- A02[1,]$PRICE
# 設定出場時間及價格
CoveryTime <- A02[nrow(A02),]$INFO_TIME
CoveryPrice <- A02[nrow(A02),]$PRICE
# 顯示交易回報
cat("BUY TIME:",OrderTime,"PRICE:",OrderPrice,"COVERY TIME:",CoveryTime,"C_PRICE:",CoveryPrice,"PROFIT",CoveryPrice-OrderPrice,'\n')

File: haohaninfo/回測範例/45-1.py
# -*- coding: UTF-8 -*-
#取I020，依照逗點分隔，並將分隔符號去除
I020 = [ line.strip('\n').split(",") for line in open('Futures_20170815_I020.csv')][1:]

#起始時間至結束時間
I020a= [ line for line in I020 if int(line[0])>9000000 and int(line[0])<11000000]


OrderTime=I020a[0][0]  #下單時間紀錄
OrderPrice=int(I020a[0][4]) #下單價格紀錄

CoverTime=I020a[-1][0]  #平倉時間紀錄
CoverPrice=int(I020a[-1][4])  #平倉時間紀錄

print "Buy OrderTime:",OrderTime," OrderPrice:",OrderPrice,
print " CoverTime:",CoverTime," CoverPrice:",CoverPrice," Profit:",CoverPrice-OrderPrice